decidim-budgets/app/commands/decidim/budgets/admin/update_projects_budget.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Budgets
+    module Admin
+      class UpdateProjectsBudget < Decidim::Command
+        include TranslatableAttributes
+
+        # Public: Initializes the command.
+        #
+        # destin_budget - the destination budget for which the projects should update to.
+        # project_ids - the project ids to update.
+        def initialize(destin_budget, project_ids)
+          @destin_budget = destin_budget
+          @project_ids = project_ids
+          @response = { selection_name: "", successful: [], errored: [], failed_ids: [] }
+        end
+
+        def call
+          return broadcast(:invalid_project_ids) if @project_ids.blank?
+          return broadcast(:invalid_project_ids) if @destin_budget.blank?
+
+          update_projects_budget
+
+          broadcast(:update_projects_budget, @response)
+        end
+
+        private
+
+        attr_reader :selection, :project_ids
+
+        def update_projects_budget
+          ::Decidim::Budgets::Project.where(id: project_ids).find_each do |project|
+            if update_allowed?(project)
+              transaction do
+                update_project_budget project
+              end
+              @response[:successful] << translated_attribute(project.title)
+            else
+              @response[:errored] << translated_attribute(project.title)
+              @response[:failed_ids] << project.id
+            end
+          end
+        end
+
+        def update_project_budget(project)
+          project.update!(
+            decidim_budgets_budget_id: @destin_budget.id
+          )
+        end
+
+        def update_allowed?(project)
+          origin_budget = project.budget
+          return false if origin_budget == @destin_budget
+
+          origin_budget.participatory_space == @destin_budget.participatory_space
```

Comment:
I have tested, and i am not 100% this is needed. I have created 2 budget components in the same space, created 1 budget for each one of it, then i have tried to move the projects. In the drop down i could not see the second component... 

maybe it should be : 

```suggestion
          origin_budget.component == @destin_budget.component
```

Reply:
We should be able to move the budgets across the different budgets component inside the same participatory space. the main issue was in [reference_budgets_for_select](https://github.com/decidim/decidim/blob/173acf5c72f30bbffa99eb24d891e5e3855dd9ed/decidim-budgets/app/helpers/decidim/budgets/projects_helper.rb#L160). I changed the method to include the other components from the same sapce